daphne/src/roles/helper.rs

```diff
@@ -201,16 +247,32 @@ pub trait DapHelper<S>: DapAggregator<S> {
                 agg_job_id_base64url: agg_job_id.to_base64url(),
             })?;
 
-        let agg_job_resp = run_vdaf_cont_round_and_aggregate(
-            self,
+        let agg_job_resp =
+            finish_agg_job_and_aggregate(self, task_id, task_config, metrics, |report_status| {
+                task_config.vdaf.handle_agg_job_cont_req(
+                    task_id,
+                    task_config,
+                    &state,
+                    report_status,
+                    &agg_job_id,
+                    &agg_job_cont_req,
+                )
+            })
+            .await?;
+
+        for transition in agg_job_resp.transitions.iter() {
```

Comment:
Reviewer note: Moved this out of `finish_agg_job_and_aggregate()` to avoid double-counting.